src/rules/no-standalone-expect.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import {
+  createRule,
+  isDescribe,
+  isExpectCall,
+  isFunction,
+  isTestCase,
+} from './tsUtils';
+
+const getBlockType = (
+  stmt: TSESTree.BlockStatement,
+): 'function' | 'describe' | 'callExpr' | false => {
+  const func = stmt.parent;
+
+  /* istanbul ignore if */
+
+  if (!func) {
+    throw new Error(
+      `Unexpected BlockStatement. No parent defined. - please file a github issue at https://github.com/jest-community/eslint-plugin-jest`,
+    );
+  }
+  // functionDeclaration: function func() {}
+  if (func.type === AST_NODE_TYPES.FunctionDeclaration) {
+    return 'function';
+  }
+  if (isFunction(func) && func.parent) {
+    const expr = func.parent;
+    // arrowfunction or function expr
+    if (expr.type === AST_NODE_TYPES.VariableDeclarator) {
+      return 'function';
+    }
+    // if it's not a variable, it will be callExpr, we only care about describe
+    if (expr.type === AST_NODE_TYPES.CallExpression && isDescribe(expr)) {
+      return 'describe';
+    }
+    return false;
```

Comment:
You don't need this `return`, since it's the same as the one outside the `if` :)